### PR TITLE
docs: Add BSD-3 license badge to README

### PR DESCRIPTION
**Pull request by exdx**
Adds a BSD-3 badge to the README. The badge is both stylish and also let's users know the license of the project right away.

image

Signed-off-by: Dan Sover <dan.sover@avalabs.org>

### CODE REVIEW

**Review by darioush — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 <p align="center">
   <a href="https://github.com/ava-labs/hypersdk/actions/workflows/unit-tests.yml"><img src="https://github.com/ava-labs/hypersdk/actions/workflows/unit-tests.yml/badge.svg" /></a>
   <a href="https://github.com/ava-labs/hypersdk/actions/workflows/static-analysis.yml"><img src="https://github.com/ava-labs/hypersdk/actions/workflows/static-analysis.yml/badge.svg" /></a>
+  <a href="https://github.com/ava-labs/hypersdk/blob/main/LICENSE"><img src="https://img.shields.io/badge/License-BSD--3-blue" /></a>
```
Comment (dhrubabasu) [inline README.md:10]:
```suggestion
  <a href="https://github.com/ava-labs/hypersdk/blob/main/LICENSE"><img src="https://img.shields.io/github/license/ava-labs/hypersdk" /></a>
```

**Review by dhrubabasu — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (joshua-kim) [inline README.md:10]: Looks like we can even use an auto-generated image that scans our repo for the correct license type, so if we ever change the license it auto-updates https://shields.io/category/license, (i.e https://img.shields.io/github/license/ava-labs/hypersdk for a default one)

**Review by joshua-kim — COMMENTED** [review]
Looks good, one comment!

File: README.md (hunk 1 above)
Comment (exdx) [reply README.md]: Done 👍

Comment (exdx) [reply README.md]: Yup good call -- I went there first but didn't see it. Updated the PR to use shields.io instead 👍

**Review by joshua-kim — APPROVED** [review]

**Pull request merged**